src/main/java/by/yahor_kulesh/dao/TicketDAO.java

```diff
@@ -1,107 +1,69 @@
-            }else return null;
-        } catch(SQLException e) {
-            System.err.println(e.getMessage());
-            return null;
-        }
+        return jdbcTemplate.query("SELECT * FROM ticket where id=?", TicketMapper.INSTANCE, id).stream().findAny().orElse(null);
     }
 
-    public ObjectArray getByUserId(UUID id) {
-        ObjectArray tickets = new ObjectArray();
-        try(
-                PreparedStatement statement = dataSource.getConnection().prepareStatement("SELECT * FROM ticket where user_id=?")
-        ) {
-            statement.setObject(1, id);
-            rs = statement.executeQuery();
-            while(rs.next()) {
-                tickets.add(getTicketFromDB());
-            }
-            return tickets;
-        } catch(SQLException e) {
-            System.err.println(e.getMessage());
-            return null;
-        }
+    @Transactional
+    public List<TicketEntity> getByUserId(UUID id) {
+        return jdbcTemplate.query("SELECT * FROM ticket where user_id=?", TicketMapper.INSTANCE, id);
     }
 
+    @Transactional
     public void update(TicketEntity ticket){
-        try(
-                PreparedStatement statement = dataSource.getConnection().prepareStatement("UPDATE ticket SET id=?,user_id=?,ticket_type=?,creation_date=? WHERE id=?")
-        ){
-            prepareTicketForStatement(ticket, statement);
-            statement.setObject(5,ticket.getId());
-            statement.executeUpdate();
-        } catch (SQLException e){
-            System.err.println(e.getMessage());
-        }
+        jdbcTemplate.update("UPDATE ticket SET id=?,user_id=?,ticket_type=?::ticket_type,creation_date=? WHERE id=?",
+                ticket.getId(),
+                ticket.getUser()==null?null:ticket.getUser().getId(),
+                ticket.getType().name(),
+                ticket.getCreationTime(),
+                ticket.getId()
+        );
     }
 
+    @Transactional
     public void deleteById(UUID id){
-        try(
-                PreparedStatement statement = dataSource.getConnection().prepareStatement("DELETE FROM ticket WHERE id=?")
-        ){
-            statement.setObject(1, id);
-            statement.execute();
-        } catch(SQLException e) {
-            System.out.println(e.getMessage());
-        }
+        jdbcTemplate.update("DELETE FROM ticket WHERE id=?", id);
     }
-    private void prepareTicketForStatement(TicketEntity ticket, PreparedStatement stat) throws SQLException {
-        stat.setObject(1,ticket.getId());
-        stat.setObject(2,ticket.getUser()==null?null:ticket.getUser().getId());
-        stat.setObject(3,ticket.getType(), Types.OTHER);
-        stat.setTimestamp(4, ticket.getCreationTime());
+
+    @Transactional
+    public void insertTicketToUser(TicketEntity ticket){
+        jdbcTemplate.update("UPDATE usr SET status=?::status_type WHERE id=?", UserStatus.ACTIVATED.name(),ticket.getUser().getId());
+        insert(ticket);
```

Comment:
do not make a call from one method in repository to another method in repository.

You can choose 1 of 2 approaches:
1. create  method with `@Transactional` in service layer and make 2 calls to 2 dao layers
2. or make 2 `jdbcTemplate.update(...)` calls inside your `insertTicketToUser` (in this case rename it more specifcallly,   for example to `createTicketAndInsertTicketToUser`), for example:
```java
 @Transactional
    public void createTicketAndInsertTicketToUser(Ticket ticket) {
        jdbcTemplate.update(YOUR_QUERY, params); // create ticket
        jdbcTemplate.update(YOUR_QUERY, params); // update user
```